lighthouse-core/computed/resource-summary.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('./computed-artifact.js');
+const NetworkRecords = require('./network-records.js');
+const MainResource = require('./main-resource.js');
+const URL = require('../lib/url-shim.js');
+
+class ResourceSummary {
+  /**
+   * @param {LH.Artifacts.NetworkRequest} record
+   * @return {LH.Budget.BudgetResourceType}
+   */
+  static determineResourceType(record) {
+    /** @type {Partial<Record<LH.Crdp.Page.ResourceType, LH.Budget.BudgetResourceType>>} */
+    const mapping = {
+      'Stylesheet': 'stylesheet',
+      'Image': 'image',
+      'Media': 'media',
+      'Font': 'font',
+      'Script': 'script',
+      'Document': 'document',
+    };
+    if (!record.resourceType) return 'other';
+    return mapping[record.resourceType] || 'other';
```

Comment:
will you add a quick test for the fallback case (`'counts unknown resource type as "other"'` or whatever)